src/components/ProductItem.jsx

```diff
@@ -46,6 +66,19 @@ const ProductItem = ({ book:
                 New Arrival
             </span>:
     null;
+
+    const handleAddToCart = async () => {
+        const productPostedToCart = await postToCart(book, token);
+    
+        if(productPostedToCart) {   
+            showToast('Book added to the cart', 'success');
+            cartDispatch({type: 'ADD_TO_CART', payload: {cartItems: productPostedToCart, error:false, loading: false}});
+            setBookInCart(true);
+        }
+        else {
+            showToast('Failed to add book to cart. Please try again later.', 'error');
```

Comment:
You can put these messages in constants

Reply:
We could if we had the same messages used for all the cases. But since the messages are different in every case, I thought it did not make sense to create constants. Maybe for the themes, it makes sense.